### ISSUE TITLE
Can't reconnect with stale localStorage.walletconnect data

### ISSUE THREAD

**Issue by Velenir** [WalletConnect/walletconnect-monorepo#193]
Good day!

I have a problem with reestablishing connection to a mobile Wallet when **localStorage** in the browser has connection information saved but the session was terminated on the side of the Wallet.

I have a minimal example [here](https://codesandbox.io/s/walletconnect-setup-48ezu) but it also happens exactly the same at https://example.walletconnect.org/ . Wallets I tested with: Trust Wallet, Gnosis Safe and https://test.walletconnect.org/
Steps to reproduce:

1. Connect to a mobile wallet
2. Close the page
3. Disconnect on the mobile wallet (page closed -> **localStorage** isn't cleared)
4. Open the page again
5. Try to connect

Connection is established (or looks established) for a brief moment and then disconnect event happens as the session data in **localStorage** is obviously stale.

What would you recommend to handle this?
After briefly browsing the source code I saw that a socket is opened in WalletConnector constructor, which causes a disconnect event if connection data is stale, so my workaround has been to wait a second, but that is obviously not ideal.

I was thinking maybe a property like `connected` but one that would return a Promise which will resolve only when WalletConnector has tried and either failed or succeeded in establishing a connection.
So that I could do
```js
const provider  = new WalletConnect({
    infuraId: "***"
  });

// immediately resolved with false if nothing in localStorage
// resolved with true if connection could be reestablished with the data from localStorage
// resolved with false if connection failed
const connected = await provider.wc.connectedPromise

await provider.enable()
```

Current behavior makes it hard to setup _autoconnection_ to WalletConnect on page load as we can't be sure the session data in **localStorage** is valid. Have to introduce an artificial delay to be sure.

**Comment by pedrouid** [comment]
@Velenir could you confirm this has been fixed with PR #206?

**Comment by jklmli** [comment]
This is still an issue - this doesn't work:
```javascript
await activate(walletconnect, undefined, true);
```
will log `Warning: Suppressed stale connector activation` due to `StaleConnectorError`.

One workaround is this:
```javascript
await activate(walletconnect, undefined, true);
await (walletconnect.walletConnectProvider as WalletConnectProvider).onDisconnect();
await activate(walletconnect, undefined, true);
```

This is a little more robust:
```javascript
const provider = walletconnect.walletConnectProvider as WalletConnectProvider;
await Promise.any([
  new Promise((resolve) => provider.onConnect(resolve)),
  provider
    .onDisconnect()
    .then(() => activate(walletconnect, undefined, true)),
]);
```

cc @pedrouid @Velenir ^
